### PR TITLE
:tada:  Linkedin Ads: support of oAuth2

### LINKED ISSUE #7709: Source Linkedin Ads: support oauth
## Tell us about the problem you're trying to solve
With the release of Airbyte Cloud, we need to start supporting Oauth for this connector, since it's the recommended way of authenticating users into a SaaS application.

If this connector doesn't support oauth already (i.e: doesn't accept a client_id and client_secret) then we need to update its spec to accept those parameters. There are two ways to do this:

If the connector already supports some auth mechanism like api_key, I suggest that this be a oneof nested inside a top-level field called "authentication":

{ authentication: { type: object oneOf: [ // api key, // oauth ] } }

If the connector only supports webflow oauth, then no changes are needed to the properties format and we will only need to add annotations.

See the connector spec reference in the docs for reference on how a oneof can be implemented.

This should be done in a backwards compatible manner i.e: users currently supplying authentication info in the config's top-level should not be impacted by this change.

## Acceptance criteria
The connector supports oauth webflow authentication with client_id/client_secret in a backwards compatible manner
Oauth properties are annotated properly. See this PR for an example

### PR DESCRIPTION
**Pull request by antixar**
## How
Add an implementation of oAuth2 logic to the connector and the Java server part

Manual result of tests:
1. Redirect to the Linkedin Ads auth windows:
image
2. Auth result:
image

## Pre-merge Checklist

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [x] Documentation updated 
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

### CODE REVIEW

File: airbyte-oauth/src/main/java/io/airbyte/oauth/flows/LinkedinAdsOAuthFlow.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Supplier;
+import org.apache.http.client.utils.URIBuilder;
+
+public class LinkedinAdsOAuthFlow extends BaseOAuth2Flow {
+
+  private static final String AUTHORIZE_URL = "https://www.linkedin.com/oauth/v2/authorization";
+  private static final String ACCESS_TOKEN_URL = "https://www.linkedin.com/oauth/v2/accessToken";
+  private static final String SCOPES = "r_ads_reporting r_ads r_basicprofile r_organization_social";
```
Comment (bazarnov) [inline airbyte-oauth/src/main/java/io/airbyte/oauth/flows/LinkedinAdsOAuthFlow.java:25]: `r_emailaddress` `r_liteprofile` should be also added to the scopes.

**Review by bazarnov — COMMENTED** [review]
Looks good! 
Couple of moments:
- Please check the comment about `scopes`
- Do you plan to add `unit` & `integration` tests for java part?

**Comment by antixar** [comment]
@bazarnov ,
I've added the[ test ](airbyte-oauth/src/test/java/io/airbyte/oauth/flows/LinkedinAdsOAuthFlowTest.java) Java part

File: airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/spec.json [hunk 2]
```diff
@@ -28,7 +22,64 @@
...
+      "credentials": {
+        "title": "Authorization Method",
+        "type": "object",
+        "oneOf": [
+          {
+            "type": "object",
+            "title": "oAuth2.0",
+            "required": ["client_id", "client_secret", "refresh_token"],
+            "properties": {
+              "credentials": {
```
Comment (midavadim) [inline airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/spec.json:35]: I would change it to auth_method or something like that.  Otherwise json path is "credentials[0]/credentials"  

 "credentials[0]/auth_method/..." - looks better

File: airbyte-oauth/src/main/java/io/airbyte/oauth/flows/LinkedinAdsOAuthFlow.java (hunk 1 above)
Comment (antixar) [reply airbyte-oauth/src/main/java/io/airbyte/oauth/flows/LinkedinAdsOAuthFlow.java:25]: Added

File: airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/spec.json (hunk 2 above)
Comment (antixar) [reply airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/spec.json]: Changed

File: docs/integrations/sources/linkedin-ads.md [hunk 3]
```diff
@@ -25,25 +25,25 @@ This Source is capable of syncing the following data as streams:
...
-| `number` | `number` | float number |
-| `integer` | `integer` | whole number |
-| `date` | `string` | FORMAT YYYY-MM-DD |
-| `datetime` | `string` | FORMAT YYYY-MM-DDThh:mm:ss |
-| `array` | `array` |  |
-| `boolean` | `boolean` | True/False |
-| `string` | `string` |  |
+| Integration Type | Airbyte Type | Notes                      |
+| :--------------- | :----------- | :------------------------- |
+| `number`         | `number`     | float number               |
```
Comment (davinchia) [inline docs/integrations/sources/linkedin-ads.md:30]: +1 for formatting.

File: tools/bin/ci_credentials.sh [hunk 4]
```diff
@@ -236,8 +236,6 @@ read_secrets source-klaviyo "$KLAVIYO_TEST_CREDS"
 read_secrets source-lemlist "$SOURCE_LEMLIST_TEST_CREDS"
 read_secrets source-lever-hiring "$LEVER_HIRING_INTEGRATION_TEST_CREDS"
 read_secrets source-looker "$LOOKER_INTEGRATION_TEST_CREDS"
-read_secrets source-linkedin-ads "$SOURCE_LINKEDIN_ADS_TEST_CREDS"
```
Comment (davinchia) [inline tools/bin/ci_credentials.sh:239]: why are these removed?

File: airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/spec.json [hunk 5]
```diff
@@ -14,12 +14,6 @@
         "description": "Date in the format 2020-09-17. Any data before this date will not be replicated.",
         "examples": ["2021-05-17"]
       },
-      "access_token": {
```
Comment (davinchia) [inline airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/spec.json:17]: can I confirm we are getting rid of the access token?

**Review by davinchia — APPROVED** [review]

File: tools/bin/ci_credentials.sh (hunk 4 above)
Comment (antixar) [reply tools/bin/ci_credentials.sh:239]: Now all secrets should be saved into GSM storage. Thus we are trying to remove all secrets from GitHub gradually

File: airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/spec.json (hunk 5 above)
Comment (antixar) [reply airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/spec.json:17]: No, both auth methods can be available. User will be able to use the new oAuth2 logic or use own access token directly
Screenshot 2021-11-18 at 00 52 58
.
